### DISCUSSION TITLE
Page path cannot contains illegal characters

### DISCUSSION THREAD

**Discussion by matrixes** [requarks/wiki#7118] [Error / Bug Report]
When creating a new page containing e.g. spaces, I cannot save the page because of the error "Page path cannot contains illegal characters" and I have found no apparent way to change the page name to allow saving the contents.

Current Version
2.5.301

**Comment by fnafox** [comment]
Spaces are not supported in page paths. If you have accidentally created one with spaces, you can delete it in the admin dashboard but I don't believe you can do anything else with it save for interacting with the database directly.

**Comment by matrixes** [comment]
Ok, then I at least didn't miss something obvious, thanks.

I would rather see that I can't create the page at all if it's not passing validation of the file name, instead of having to figure out in hindsight that I've entered non supported characters and have no way of fixing that apart from recreating the page.

(Now I just need to figure out how to convert this error/bug report into a feature request/suggestion. If nothing else it's a UX issue as it stands.)

**Comment by heinrich-foto** [comment]
The validation need to be performed in the dialogue for selecting a path. Actually the validation only performs in the site settings window. `Invalid path. Please ensure it does not contain special characters, or begin/end in a slash or hashtag string.`

In this case I found a bug in the validation step. If I insert a `%20`, I can save the site and internally create it, but it is not accessible. 

in the following dialogue, no validation gets performed. 
<img width="1017" alt="Bildschirmfoto 2025-03-06 um 11 35 31" src="https://github.com/user-attachments/assets/b2e81fb3-9922-447b-a903-ab5a4cbaf5a9" />
in the following dialogue, validation gets performed and prevent saving the site, as expected.
<img width="1022" alt="Bildschirmfoto 2025-03-06 um 11 35 40" src="https://github.com/user-attachments/assets/b18d306c-65ce-4872-8e63-ed9b208ef33e" />


We need validation in the first step also. And the validation need to take care of escaped illegal chars.

**Reply by heinrich-foto** [reply]
the % sign is included in the `const filenamePattern = /^(?![\#\/\.\$\^\=\*\;\:\&\?\(\)\[\]\{\}\"\'\>\<\,\@\!\%\`\~\s])(?!.*[\#\/\.\$\^\=\*\;\:\&\?\(\)\[\]\{\}\"\'\>\<\,\@\!\%\`\~\s]$)[^\#\.\$\^\=\*\;\:\&\?\(\)\[\]\{\}\"\'\>\<\,\@\!\%\`\~\s]*$/` the error message appears, but the save action is not prevented.
